### ISSUE TITLE
Generated setup hooks documentation

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixpkgs#197070]
### Describe the bug
Setup hooks are an important concept providing complex behaviour. Documentation is typically close to the code. We could use the longDescription as docstring and add an identifier/tag to hooks so we can recognize them. Then we can find them. Also on the website we could highlight hooks.

### Steps To Reproduce
1. ...
2. ...
3. ...

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

**Comment by FRidh** [comment]
@NixOS/documentation-team does the website support any markup inside a `longDescription`?

**Comment by fricklerhandwerk** [comment]
> does the website support any markup inside a longDescription?

@FRidh yes, search.nixos.org renders CommonMark: https://github.com/NixOS/nixos-search/pull/539

I don't really understand your proposal, though. Could you show a code sample of how you envision it to work?
